### ISSUE TITLE
Exported result value contains stdout from previous run when commands don't change

### ISSUE THREAD

**Issue by jtackaberry** [loafoe/terraform-provider-ssh#30]
When I use a `file` block to lay down a script and then invoke that script from `commands`, upon subsequent modification of the file contents the exported `result` becomes stale and contains the output from the previous invocation, not the current one.

If I mutate `commands`, then `result` is properly updated, but when `commands` is unchanged and only `file` changes, `result` is exported with the previous run's output.

Consider this excerpt:

```terraform
resource "ssh_resource" "init" {
  host              = digitalocean_droplet.green.ipv4_address
  user              = "root"
  private_key       = tls_private_key.tf.private_key_pem

  file {
    # Modifying only this line reproduces the problem
    content     = "echo one"
    destination = "/tmp/run.sh"
    permissions = "0700"
  }
  commands = [
    "/tmp/run.sh",
  ]
}

output "stdout" {
  value = ssh_resource.init.result
}
```

Given the above, first run is fine.  Then if you update the `content` line above and reapply, the `stdout` output is unchanged, however the new version of `run.sh` is definitely executed.  Running `terraform apply` yet one more time refreshes the output, even though nothing else has changed.

**Comment by loafoe** [comment]
hi @jtackaberry , thanks for reporting and sorry for the delay. This is indeed a bug and it can be solved by adding a `CustomizeDiff` method that tells Terraform the `result` will change due to a `file` or `commands` mutation ahead of time. The upcoming version will fix this, stay tuned

**Comment by loafoe** [comment]
should be fixed in `v2.1.0`
